smt/surrogate_models/kpls.py

```diff
@@ -55,3 +68,69 @@ def _componentwise_distance(self, dx, opt=0, theta=None, return_derivative=False
             return_derivative=return_derivative,
         )
         return d
+
+    def _train(self):
+        """
+        Train the model
+        """
+        # outputs['sol'] = self.sol
+
+        if self.options["eval_n_comp"]:
+            eval_comp_treshold = self.options["eval_comp_treshold"]
```

Comment:
The whole if branch could go in a specific method `estimate_number_of_components` and have specific docstring that explain that `self.options[n_comp]` value from user is ignored and replaced by an estimated one wrt Wold's R criterion.